data_source.go

```diff
@@ -345,19 +367,18 @@ func (ds *AnySource) PrepareRun() error {
 		dsp.stream.voltsPerArb = vpa[channelNum]
 		ds.processors[channelNum] = dsp
 
-		// TODO: don't just set these to arbitrary values
+		ts := tsptrs[channelNum]
+		if ts == nil {
+			ts = &defaultTS
+		}
+		dsp.TriggerState = *ts
+
+		// TODO: don't just set decimation and samples/presamples to arbitrary values
 		dsp.Decimate = false
-		dsp.DecimateLevel = 3
+		dsp.DecimateLevel = 2
 		dsp.DecimateAvgMode = true
```

Comment:
Why is Decimate on by default?

Reply:
It isn't:

`dsp.Decimate = false`

Reply:
But then why set DecimateLevel at all? Should it just be set to the zero value, and when someone turns it on they can set the values they want.